pulumi/github/repos/pipelines/testpipeline.py

```diff
@@ -0,0 +1,212 @@
+        ),
+        # 'branch_master_required_ci' => 'master branch: minimum set of CI tests must pass',
+        required_status_checks=github.RepositoryRulesetRulesRequiredStatusChecksArgs(
+            required_checks=REQUIRED_CI_CHECKS,
+            strict_required_status_checks_policy=True,
+        ),
+    ),
+    target="branch",
+    opts=pulumi.ResourceOptions(protect=True),
+)
+# TODO 'branch_dev_strict_updates' => 'dev branch: do not require branch to be up to date before merging',
+ruleset_branch_dev_testpipeline = github.RepositoryRuleset(
+    f"ruleset_branch_dev_{NAME}",
+    # 'branch_dev_enforce_admins' => 'dev branch: do not enforce rules for admins',
+    bypass_actors=[
+        github.RepositoryRulesetBypassActorArgs(
+            actor_id=CORE_TEAM_ID,
+            actor_type="Team",
+            bypass_mode="always",
+        ),
+        github.RepositoryRulesetBypassActorArgs(
+            actor_id=MAINTAINERS_TEAM_ID,
+            actor_type="Team",
+            bypass_mode="always",
+        ),
+    ],
+    conditions=github.RepositoryRulesetConditionsArgs(
+        ref_name=github.RepositoryRulesetConditionsRefNameArgs(
+            excludes=[],
+            includes=["refs/heads/dev"],
+        ),
+    ),
+    enforcement="active",
+    name="dev",
+    repository=NAME,
+    rules=github.RepositoryRulesetRulesArgs(
+        deletion=True,
+        non_fast_forward=True,
+        pull_request=github.RepositoryRulesetRulesPullRequestArgs(
+            dismiss_stale_reviews_on_push=False,  # 'branch_dev_stale_reviews' => 'dev branch: reviews not marked stale after new commits',
+            require_code_owner_review=False,  # 'branch_dev_code_owner_reviews' => 'dev branch: code owner reviews not required',
+            # TODO require_last_push_approval=True,
+            required_approving_review_count=1,  # 'branch_dev_required_num_reviews' => 'dev branch: 1 review required',
+            # TODO required_review_thread_resolution=True,
+        ),
+        # 'branch_dev_required_ci' => 'dev branch: minimum set of CI tests must pass',
+        required_status_checks=github.RepositoryRulesetRulesRequiredStatusChecksArgs(
+            required_checks=REQUIRED_CI_CHECKS,
+            strict_required_status_checks_policy=True,
+        ),
+    ),
+    target="branch",
+    opts=pulumi.ResourceOptions(protect=True),
+)
+# TODO Double check
+# Template branch protection https://github.com/nf-core/website/blob/33acd6a2fab2bf9251e14212ce731ef3232b5969/public_html/pipeline_health.php#L509
+ruleset_branch_template_testpipeline = github.RepositoryRuleset(
+    f"ruleset_branch_TEMPLATE_{NAME}",
+    bypass_actors=[
+        github.RepositoryRulesetBypassActorArgs(
+            actor_id=CORE_TEAM_ID,
+            actor_type="Team",
+            bypass_mode="always",
+        )
+        # TODO 'branch_template_restrict_push' => 'Restrict push to TEMPLATE to @nf-core-bot',
```

Comment:
I think we don't need this, everyone can run `nf-core sync` manually with their github user

Reply:
Just going for 1 to 1 with pipeline_health.

If I got that wrong, cool, but I vote we give our best attempt to get it the same, then make a PR to update those changes we can get rid of to "self-document" the change.